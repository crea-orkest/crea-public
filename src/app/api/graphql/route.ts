import { NextResponse } from 'next/server'
import { makeExecutableSchema } from '@graphql-tools/schema'
import { GraphQLError, graphql } from 'graphql'
import getConfig from 'next/config'

const { publicRuntimeConfig } = getConfig()

const typeDefs = `
  type Query {
    publicRuntimeConfig: [PublicRuntimeConfig]
  }

  type PublicRuntimeConfig {
    googleMapsApiKey: String!
  }
`

const resolvers = {
  Query: {
    publicRuntimeConfig: () => [
      {
        googleMapsApiKey: String(publicRuntimeConfig.googleMapsApiKey),
      },
    ],
  },
}

const schema = makeExecutableSchema({ typeDefs, resolvers })

export async function POST(request: Request) {
  const data = await request.json()
  try {
    const result = await graphql({
      schema,
      source: data.query,
    })
    return NextResponse.json(result)
  } catch (error: unknown) {
    if (error instanceof Error) {
      return NextResponse.json(new GraphQLError(error.message))
    }
    return NextResponse.json(new GraphQLError('random error'))
  }
}
