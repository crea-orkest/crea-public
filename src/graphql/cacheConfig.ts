import type { InMemoryCacheConfig } from "@apollo/client";
import { createFragmentRegistry } from "@apollo/client/cache";
import { identifiable } from "./fragments/identifiable";
import { seo } from "./fragments/seo";
import { tag } from "./fragments/tag";

export const cacheConfig: InMemoryCacheConfig = {
  fragments: createFragmentRegistry(identifiable, seo, tag),
  typePolicies: {
    ConcertRecord: {
      fields: {
        poster: {
          merge: true,
        },
      },
    },
  },
};
