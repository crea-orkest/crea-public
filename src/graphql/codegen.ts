import { CodegenConfig } from "@graphql-codegen/cli";

const config: CodegenConfig = {
  schema: [
    {
      [`https://graphql.contentful.com/content/v1/spaces/pwuxt9271wes/environments/${process.env.CONTENTFUL_ENVIRONMENT}`]:
        {
          headers: {
            Authorization: `Bearer ${process.env.CONTENTFUL_PREVIEW_TOKEN}`,
          },
        },
    },
  ],
  documents: ["./src/graphql/**/*.ts"],
  generates: {
    "./src/graphql/generated/types.ts": {
      plugins: ["typescript", "fragment-matcher", "typescript-operations"],
      config: { withHooks: true },
      presetConfig: {
        gqlTagName: "gql",
      },
    },
  },
  hooks: { afterAllFileWrite: ["prettier --write"] },
};

export default config;
