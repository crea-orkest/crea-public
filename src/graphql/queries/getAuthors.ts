import gql from "graphql-tag";

import { authorFragment } from "../fragments/author";

export const GET_AUTHORS = gql`
  query getAuthors($limit: Int, $skip: Int) {
    authorCollection(
      order: [sys_publishedVersion_ASC]
      limit: $limit
      skip: $skip
    ) {
      limit
      skip
      total
      items {
        ...authorFragment
      }
    }
  }
  ${authorFragment}
`;
