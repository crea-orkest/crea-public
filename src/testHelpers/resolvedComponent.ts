export const resolvedComponent = async <T>(
  Component: Function,
  props: T
): Promise<() => JSX.Element> => {
  const ComponentResolved = await Component(props);
  return () => ComponentResolved;
};
