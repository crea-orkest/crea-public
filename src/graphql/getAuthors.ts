import { contentfulClient } from "./contentfulClient";
import { GetAuthorsQuery, GetAuthorsQueryVariables } from "./generated/types";
import { GET_AUTHORS } from "./queries/getAuthors";

export const getAuthors = async ({ limit, skip }: GetAuthorsQueryVariables) => {
  try {
    const { data, loading, errors } =
      await contentfulClient.query<GetAuthorsQuery>({
        query: GET_AUTHORS,
        variables: {
          limit,
          skip,
        },
      });
    return { data, loading, errors };
  } catch (errors: unknown) {
    return { data: null, loading: false, errors };
  }
};
