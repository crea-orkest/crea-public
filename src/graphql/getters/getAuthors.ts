import { GET_AUTHORS } from "graphql/queries/getAuthors";
import { authorsFormatter } from "../formatters/authorsFormatter";
import {
  GetAuthorsQuery,
  GetAuthorsQueryVariables,
} from "../generated/graphql";
import { getClient } from "../gqlClient";

export const getAuthors = async ({
  first,
  skip,
  order,
}: GetAuthorsQueryVariables) => {
  const client = getClient();

  try {
    const { data, loading, errors } = await client.query<GetAuthorsQuery>({
      query: GET_AUTHORS,
      variables: { first, skip, order },
    });
    return { data: authorsFormatter(data), loading, errors };
  } catch (errors) {
    if (errors instanceof Error) console.log(errors.message);

    return { data: null, loading: false, errors };
  }
};
