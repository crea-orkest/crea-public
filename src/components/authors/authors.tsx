import type { PersonModelOrderBy } from 'graphql/generated/graphql'
import React from 'react'
import { getAuthors } from 'graphql/getters/getAuthors'

export interface Props {
  skip: number
  first: number
  order: PersonModelOrderBy
}

export const Authors: React.FC<Props> = async ({ skip, first, order }) => {
  const { data } = await getAuthors({ skip, first, order })
  if (!data) return null

  return (
    <ul>
      {data.map((item) => (
        <li key={item.id}>{item.name}</li>
      ))}
    </ul>
  )
}
