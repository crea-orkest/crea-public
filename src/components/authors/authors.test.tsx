import { Authors } from './authors'
import { PersonModelOrderBy } from '../../graphql/generated/graphql'
import type { Props } from './authors'
import React from 'react'
import { getAuthors } from '../../graphql/getters/getAuthors'
import { resolvedComponent } from '../../utils/testHelpers/resolvedComponent'
import { render, screen } from '@testing-library/react'

jest.mock('../../graphql/getters/getAuthors', () => {
  const originalModule = jest.requireActual('../../graphql/getters/getAuthors')
  return {
    __esModule: true,
    ...originalModule,
    getAuthors: jest.fn(),
  }
})

const getAuthorsMock = jest.mocked(getAuthors)

describe('Authors component', () => {
  it('shows all the data', async () => {
    getAuthorsMock.mockResolvedValueOnce({
      data: [
        {
          id: 'id',
          name: 'name',
          description: 'role',
        },
      ],
      error: undefined,
    })

    const Resolved = await resolvedComponent<Props>(Authors, {
      skip: 0,
      first: 1,
      order: PersonModelOrderBy.CreatedAtAsc,
    })

    const { container } = render(<Resolved />)

    expect(container).toMatchSnapshot()
    expect(screen.getByText('name')).toBeTruthy()
  })
})
