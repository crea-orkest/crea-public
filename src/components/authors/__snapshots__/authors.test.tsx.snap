// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Authors component shows all the data 1`] = `
<div>
  <ul>
    <li>
      name
    </li>
  </ul>
</div>
`;
